File: watch_hub_frontend/lib/models/cart_item.dart
import 'package:watch_hub_frontend/models/watch.dart';

class CartItem {
  final String id;
  final Watch watch;
  final String color;
  final String size;
  int quantity;

  CartItem({
    required this.id,
    required this.watch,
    required this.color,
    required this.size,
    this.quantity = 1,
  });

  double get totalPrice => watch.discountedPrice * quantity;
}

File: watch_hub_backend/app/Models/CartItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CartItem extends Model
{
    use HasFactory;

    protected $fillable = [
        'cart_id',
        'variant_id',
        'quantity',
    ];

    protected $casts = [
        'quantity' => 'integer',
    ];

    // Relationships
    public function cart()
    {
        return $this->belongsTo(Cart::class);
    }

    public function variant()
    {
        return $this->belongsTo(WatchVariant::class);
    }

    // Helper Methods
    public function getSubtotalAttribute()
    {
        $watch = $this->variant->watch;
        $price = $watch->final_price + $this->variant->additional_price;
        return $price * $this->quantity;
    }
}
